### PR TITLE
Expose the possibility of setting the json settings and more advanced scenarios when needing to override the readers and writers

### LINKED ISSUE #262: Saga data serialization. Provide custom JsonSerializerSettings
I've noticed that both SqlPersistence and ServiceFabricPersistence allow custom JsonSerializerSettings but AzureStoragePersistence does not. Is this something worth considering?

```
var jsonSerializerSettings = new JsonSerializerSettings
{
    ContractResolver = new PrivateSetterAndCtorContractResolver()
};

var persistence = endpoint.UsePersistence<AzureStoragePersistence>().ConnectionString("UseDevelopmentStorage=true");
var sagaSettings = persistence.SagaSettings();
sagaSettings.JsonSettings(jsonSerializerSettings);
```

### PR DESCRIPTION
**Pull request by danielmarbach**
Closes #262

### CODE REVIEW

File: src/NServiceBus.Persistence.AzureTable/SagaPersisters/ConfigureAzureSagaStorage.cs [hunk 1]
```diff
@@ -46,6 +49,50 @@ public static class ConfigureAzureSagaStorage
             return config;
         }
 
+        /// <summary>
+        /// Overrides the default settings used by the saga property serializer applicable for complex property types that are not supported by default with tables.
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureTable/SagaPersisters/ConfigureAzureSagaStorage.cs:53]:
```suggestion
        /// Overrides the default settings used by the saga property serializer used for complex property types serialization that is not supported by default with tables.
```

File: src/NServiceBus.Persistence.AzureTable/SagaPersisters/ConfigureAzureSagaStorage.cs [hunk 2]
```diff
@@ -46,6 +49,50 @@ public static class ConfigureAzureSagaStorage
...
+            Guard.AgainstNull(nameof(jsonSerializerSettings), jsonSerializerSettings);
+
+            var settings = config.GetSettings();
+            settings.Set(WellKnownConfigurationKeys.SagaJsonSerializer, JsonSerializer.Create(jsonSerializerSettings));
+
+            return config;
+        }
+
+        /// <summary>
+        /// Overrides the reader creator to influence custom data reading scenarios.
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureTable/SagaPersisters/ConfigureAzureSagaStorage.cs:67]:
```suggestion
        /// Overrides the reader creator to customize data deserialization.
```

File: src/NServiceBus.Persistence.AzureTable/SagaPersisters/ConfigureAzureSagaStorage.cs [hunk 3]
```diff
@@ -46,6 +49,50 @@ public static class ConfigureAzureSagaStorage
...
+
+            var settings = config.GetSettings();
+
+            settings.Set(WellKnownConfigurationKeys.SagaReaderCreator, readerCreator);
+
+            return config;
+        }
+
+        /// <summary>
+        /// Overrides the writer creator to influence custom data writing scenarios.
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.AzureTable/SagaPersisters/ConfigureAzureSagaStorage.cs:82]:
```suggestion
        /// Overrides the writer creator to customize data serialization.
```

**Review by SeanFeldman — COMMENTED** [review]
Minor xml documentation suggestions

**Review by yvesgoeleven — APPROVED** [review]

**Review by SeanFeldman — APPROVED** [review]

**Pull request merged**